Assets/Scripts/Candy.cs

```diff
@@ -29,18 +29,18 @@ public class Candy : MonoBehaviour, ICandy
     public float SizeY { get; set; }
     public float PosX { get ; set ; }
     public float PosY { get ; set ; }
-    public int PosInArrayI { get { return posInArrayI; } set { posInArrayI = value; } }
-    public int PosInArrayJ { get { return posInArrayJ; } set { posInArrayJ = value; } }
+    public int PosInArrayI { get { return posInArrayI; } set { posInArrayI = (int)value; } }
+    public int PosInArrayJ { get { return posInArrayJ; } set { posInArrayJ = (int)value; } }
 
     private GameSettings gameSettings;
 
 
     public void ResetProperties()
     {
-        PosX = default;
-        PosY = default;
-        PosInArrayI = default;
-        PosInArrayJ = default;
+        PosX = 0;
+        PosY = 0;
+        PosInArrayI = 0;
+        PosInArrayJ = 0;
```

Comment:
Here it makes sense to reset the candy type as well. I couldn't find where exactly you are changing the CandyType. It looks like it never gets set 🤔 or am I wrong?

Reply:
or what you should do, is to set the candy type when you dequeue it from the pool and want to use it.